compiler/rustc_parse/src/parser/diagnostics.rs

```diff
@@ -1933,7 +1933,19 @@ impl<'a> Parser<'a> {
         }
         match self.parse_expr_res(Restrictions::CONST_EXPR, None) {
             Ok(expr) => {
-                if token::Comma == self.token.kind || self.token.kind.should_end_const_arg() {
+                // Find a mistake like `MyTrait<Assoc == S::Assoc>`.
+                if token::EqEq == snapshot.token.kind {
+                    err.span_suggestion(
+                        snapshot.token.span,
+                        "replace `==` with `=`",
+                        "=".to_string(),
+                        Applicability::MaybeIncorrect,
+                    );
+                    let value = self.mk_expr_err(expr.span);
```

Comment:
Shouldn't this be `self.mk_expr_err(start.to(expr.span))`? We'd want this to cover the whole `Foo == Bar` instead of just `Bar`.